src/cfnlint/custom_rules.py

```diff
@@ -0,0 +1,52 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+# pylint: disable=W0108
+# pylint: disable=W0622
+import logging
+import cfnlint
+import cfnlint.customRules.Operators
+
+LOGGER = logging.getLogger(__name__)
+Operator = {'EQUALS': lambda x, y: cfnlint.customRules.Operators.equalsOp(x, y),
+            'PLACEHOLDER': lambda x, y: LOGGER.debug('Placeholder Op')}
+
+def check(filename, template):
+    """ Process custom rule file """
+    matches = []
+    with open(filename) as customRules:
+        count = 1
+        for line in customRules:
+            LOGGER.debug('Processing Custom Rule Line %d', count)
+            line = line.replace('"', '')
+            rule = line.split(' ')
+            if len(rule) == 4 and rule[0][0] != '#':
+                resource_type = rule[0]
+                operator = rule[2]
+                try:
+                    result = Operator[rule[2]](rule, template.get_resource_properties([resource_type]))
```

Comment:
Bit of a code smell that when I was inspecting the code for `equalsOp` I had to go back here to see what parameter it was receiving.. maybe it would be more readable if you passed in the `template` directly and let the operator do whatever it wants with it.. and if you think about it, an operator is a rule that acts on a template so it makes sense conceptually

Reply:
I think that code definitely could be more readable. Per the earlier process diagram, I'm working to isolate components from each other, like in this case with the split between the identification of relevant properties, and the actual application of the operator onto said properties.

I'm would appreciate your feedback on how things look on the latest version which has been cleaned up a fair bit with the implementation of your later suggestions!